unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/Debugging/Performance/LinealBufferHiccupCounter.cs

```diff
@@ -1,67 +1,41 @@
-﻿namespace DCL.FPSDisplay
+﻿using MainScripts.DCL.WorldRuntime.Debugging.Performance;
+
+namespace DCL.FPSDisplay
 {
     public class LinealBufferHiccupCounter
     {
-        private readonly LinealBufferFPSCounter counter = new LinealBufferFPSCounter();
-        private int hiccupsCountInBuffer = 0;
-        private float hiccupsSum = 0.0f;
-        private float bufferCount = 0.0f;
-
-        public int HiccupsCountInBuffer
-        {
-            get => hiccupsCountInBuffer;
-            private set => hiccupsCountInBuffer = value;
-        }
-
-        public float HiccupsSum
-        {
-            get => hiccupsSum;
-            set => hiccupsSum = value;
-        }
+        private readonly IProfilerRecordsService profilerRecordsService;
 
-        public float CurrentFPSCount()
-        {
-            return counter.CurrentFPSCount();
-        }
+        public int HiccupsCountInBuffer { get; private set; }
+        public float HiccupsSum { get; private set; }
+        public float TotalSeconds { get; private set; }
 
-        public float GetTotalSeconds()
+        public LinealBufferHiccupCounter(IProfilerRecordsService profilerRecordsService)
         {
-            return bufferCount;
+            this.profilerRecordsService = profilerRecordsService;
         }
 
         public void AddDeltaTime(float valueInSeconds)
         {
-            if (IsHiccup(counter.values[counter.end]))
+            float lastFrameTimeInSec = profilerRecordsService.LastFrameTimeInSec;
+
+            if (IsHiccup(lastFrameTimeInSec))
             {
-                hiccupsCountInBuffer -= 1;
-                hiccupsSum -= counter.values[counter.end];
+                HiccupsCountInBuffer -= 1;
+                HiccupsSum -= lastFrameTimeInSec;
```

Comment:
Please add tests to ensure this works exactly like the previous one. I'd even suggest to revert this specific piece, the new implementation is not doing the same at plain sight (can't confirm since tests are no longer there). The previous one used a O(1) circular buffer and removed the oldest measurement from the buffer. This one is removing the newest.

The correct behavior of this buffer is crucial for all of our performance dashboards, unless it has serious problems I'd prevent touching it. If we change the measurement method suddenly all of our new data is not comparable with the previous.

The rest of the PR looks sensible though

Reply:
after our discussion and small investigation, I realized that these measurements are not used in the backend. Because
`counter.values` array is never send to the backend and we send `PerformanceMetricsController.encodedSamples` array instead. As for hiccups, as you mentioned, we calculate them differently on the backend (however, we still send these ones 🤔 maybe to support `old_hiccups` chart, I don't know). Anyway, I brought back old `counter.values` array for hiccups counting.